addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -499,6 +499,289 @@ registry.slider = publicWidget.Widget.extend({
     },
 });
 
+registry.backgroundVideo = Animation.extend({
+    selector: '.o_video_background',
+    jsLibs: [
+        '/website/static/lib/YTPlayer/jquery.mb.YTPlayer.js',
+    ],
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    start: function () {
+        this._startVideo();
+        return this._super.apply(this, arguments);
+    },
+
+    /**
+     * @override
+     */
+    destroy: function () {
+        this._stopVideo();
+        this._super.apply(this, arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * When user provides video this method will update the video type according 
+     * to the input.
+     *
+     * @private
+     */
+    _updateVideoType: function () {
+        var regexUrl = '((?:https?:)?//([^\\s\'"<>\\\\/]+)/([^\\s\'"<>\\\\]+))';
+        var match = this.src.match(new RegExp('\\ssrc=[\'"]?' + regexUrl));
+        match = match || this.src.match(new RegExp('^\\s*' + regexUrl));
+        if (!match) {
+            this.videoType = "image";
+            this.src = "";
+            return;
+        }
+
+        var url = match[1];
+        var domain = match[2];
+        var path = match[3];
+
+        match = undefined;
+
+        var servicesPrefix = {
+            youtube: 'https://youtu.be/',
+            vimeo: 'https://vimeo.com/',
+            dailymotion: 'http://dai.ly/',
+        };
+
+        if (/\.youtube(-nocookie)?\./.test(domain)) {
+            this.videoType = 'Youtube';
+            match = path.match(/^(?:embed\/|watch\?v=)?([^\/?&#]+)/i);
+        } else if (domain === "youtu.be") {
+            this.videoType = 'Youtube';
+            match = path.match(/^([^\/?&#]+)/);
+        } else if (_.str.include(domain, "vimeo.")) {
+            this.videoType = 'vimeo';
+            match = path.match(/^(?:video\/)?([^?&#]+)/i);
+        } else if (_.str.include(domain, ".dailymotion.")) {
+            this.videoType = "dailymotion";
+            match = path.match(/(?:embed\/)?(?:video\/)?([^\/?&#_]+)/i);
+        } else if (domain === "dai.ly") {
+            this.videoType = "dailymotion";
+            match = path.match(/^([^\/?&#]+)/);
+        }
+
+        if (match) {
+            this.src = servicesPrefix[this.videoType] + match[1];
+        } else if (!/\ssrc=/.test(this.src)) {
+            this.src = url;
+            this.videoType = 'html5';
+        } else {
+            this.videoType = 'other';
+        }
+
+        this.$target.data('video-type', this.videoType);
+    },
+
+    /**
+     * When user provides video this method will be called first.
+     *
+     * @private
+     */
+    _startVideo: function () {
+        var self = this;
+        var videoUrl;
+
+        if (!self.videoType || self.src !== self.$target.attr('src')) {
```

Comment:
Don't use `self` when you can use `this`

Reply:
Hello @qsm-odoo, `_updateVideoType` method is as it is which was present in the media dialog snippet for checking the video type and all other changes are done. I have squashed all the commits and updated the code as per your review. :)

Reply:
Yes but we should improve the method that was moved from the themes repo, the goal is to share the code and to have clean code. Please try checking :)